test/unit/test_vec.c

```diff
@@ -850,6 +850,74 @@ static bool test_vec_upper_bound(void) {
 	mu_end;
 }
 
+static int compare_S(const S *a, const S *b) {
+	if (*a->y < *b->y) {
```

Comment:
probably worth caching ay and by into local const vars to avoid referencing them twice

Reply:
Yes, but this is unit test code so I'm not worried about it 🙈